File: app/src/main/java/com/ivanloy/nutricards/ds/Deck.java
package com.ivanloy.nutricards.ds;

import com.ivanloy.nutricards.game_elements.FoodCard;

public class Deck extends Stequeue<FoodCard> {

    public void putCardOnTop(FoodCard card){
        super.push(card);
    }

    public void putCardOnBottom(FoodCard card){
        super.enqueue(card);
    }

    public FoodCard drawCard(){
        return super.pop();
    }

}

File: app/src/main/java/com/ivanloy/nutricards/ds/Deck.kt
package com.ivanloy.nutricards.ds

import com.ivanloy.nutricards.gameelements.FoodCard

class Deck : Stequeue<FoodCard>() {

    fun putCardOnTop(card: FoodCard) {
        super.push(card)
    }

    fun putCardOnBottom(card: FoodCard) {
        super.enqueue(card)
    }

    fun drawCard(): FoodCard {
        return super.pop()
    }

}
